[jiahui0309] iP

jiahui0309 opened the pull request:
### Dino

> “The best way to predict your future is to create it.” —Abraham Lincoln

Dino is a chatbot that allows you to track your tasks.
It's:

- text-based
- user-friendly 
- ~FAST~ _SUPER_ FAST to user

All you need to do is:

1. download it [here](https://github.com/jiahui0309/ip)
2. change the directory to where your jar file is
3. run 'java -jar <name of the jar file>.jar'
4. add your tasks
5. and you're done! 🥳

And it is **_FREE_**! 😩

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jiahui0309 committed:
Level 0 - Rename, Greet, Exit

jiahui0309 committed:
Level 1 - Echo

jiahui0309 committed:
Level-2: Add, List

jiahui0309 committed:
Implementing Task class

jiahui0309 committed:
Level-3: Mark as Done

jiahui0309 committed:
Level-4: ToDo, Event, Deadline

jiahui0309 committed:
ran runtest.sh

jiahui0309 committed:
edited Dino.java for better readability

jiahui0309 committed:
Level-5: created Exceptions to handle errors

jiahui0309 committed:
Level-6: feature for deleting tasks

jiahui0309 committed:
minor edits for Level-6

jiahui0309 committed:
Create Enum classes

jiahui0309 committed:
Save tasks after exiting

jiahui0309 committed:
Merge branch 'branch-level-7'

jiahui0309 committed:
Merge branch-level-7 for Level-7

jiahui0309 committed:
Add LocalDate class

jiahui0309 committed:
Filter deadlines and events on a specific date

jiahui0309 committed:
Merge branch-level-8 for Level 8

jiahui0309 committed:
Implement OOP for the classes

jiahui0309 committed:
Refactor classes into packages

jiahui0309 committed:
Merge A-Packages

jiahui0309 committed:
Merge Gradle support provided

jiahui0309 committed:
Set up gradle support

jiahui0309 committed:
Add JUnit tests

jiahui0309 committed:
Add more JUnit tests

jiahui0309 committed:
Export app as an executable JAR file

jiahui0309 committed:
Add JavaDoc to methods

jiahui0309 committed:
Add Coding Standards to methods

jiahui0309 committed:
Find tasks with specified keyword

jiahui0309 committed:
Updated checkstyle

jiahui0309 committed:
Add JavaFX application

jiahui0309 committed:
Fix bugs in Level-10

jiahui0309 committed:
Merge branch-level-10 for Level 10

jiahui0309 committed:
Made code fixes

jiahui0309 committed:
Merge Git Standards for coding standards

jiahui0309 committed:
Add JavaFX tutorial part 1-3

jiahui0309 committed:
Add JavaFX tutorial part 4

jiahui0309 committed:
Resolve merge conflicts

jiahui0309 committed:
Edit Parser class

jiahui0309 committed:
Minor fixes for level-10

Darylgolden commented on src/main/java/dino/command/Parser.java:
```diff
@@ -0,0 +1,196 @@
+
+            result.append("Tasks for ").append(date).append(":\n");
+
+            tasks.getTaskList().stream()
```
I like how the long stream chains are split into multiple lines.

Darylgolden commented on src/main/java/dino/command/Parser.java:
```diff
@@ -0,0 +1,196 @@
+     */
+    public String parseStringToNum(String time) {
+        time = time.trim();
+        DateTimeFormatter dateOnlyFormatter = DateTimeFormatter.ofPattern("MMM dd yyyy", Locale.ENGLISH);
```
Perhaps these locales can be stored as a static variable instead? This would allow changes to be made easily if necessary.

Darylgolden commented on src/main/java/dino/command/TaskList.java:
```diff
@@ -0,0 +1,110 @@
+    /**
+     * Removes the Task from the TaskList.
+     *
+     * @param taskNum The index of the Task to be removed.
```
I think it could be more clear that the index is 1-indexed.

Darylgolden commented on src/main/java/dino/task/Task.java:
```diff
@@ -0,0 +1,70 @@
+     * Returns a boolean representation whether Task contains the keyword.
+     *
+     * @param keyword String value input.
+     * @return        True if description contains keyword, else False.
```
I'm not sure if the large spacing here is appropriate?

Darylgolden reviewed (commented):
I think this PR LGTM! I could only find very minor nitpicks; I think you've done a great job following the specified coding standard throughout the entire codebase.

jiahui0309 committed:
Create gradle.yml

jiahui0309 committed:
Update gradle.yml

jiahui0309 committed:
Merge branch 'master' of https://github.com/jiahui0309/ip

jiahui0309 committed:
Update classes for CI runs

jiahui0309 committed:
Update Parser.java

jiahui0309 committed:
Merge branch 'master' of https://github.com/jiahui0309/ip

jiahui0309 committed:
Build new jar file

jiahui0309 committed:
Current code does not have assert feature.

Usage of assert feature can document important assumptions that should hold at various points in the code.

Let's,
* add the assert feature

jiahui0309 committed:
Refactor code for improved abstraction: Reorganize Parsing Logic

The code currently has differing levels of abstraction, especially when parsing user responses.

To enhance clarity and maintainability, we should extract common behaviors and utilize an abstract class to categorize the various types of commands.

Let's:
* Create an abstract class for handling commands.

jiahui0309 committed:
Refactor code for improved abstraction: Removed deep nesting

The previous implementation of the loadTasksFromFile() method contained deep nesting, as there was noticeable arrowhead style code. This commit addresses this issue by refactoring the method to reorganize the parsing logic into smaller, more manageable methods.

Changes Made:
* Introduced a parseTaskData() method to handle the parsing of task data and addition of parsed tasks to the TaskList.
* Created separate methods (parseTodoTask(), parseDeadlineTask(), and parseEventTask()) to parse different types of tasks from the task data.
* Updated the loadTasksFromFile() method to call these new parsing methods, reducing the nesting level.

Benefits:
* Improved code readability: The reorganized parsing logic makes the code easier to understand and modify.
* Reduced nesting: This enhances code structure and clarity.
* Better separation of tasks: Each parsing method is responsible for parsing a specific type of task.

jiahui0309 committed:
Add Javadocs for new classes and methods

jiahui0309 committed:
Merge pull request #2 from jiahui0309/branch-A-Assertions

Current code does not have assert feature.

jiahui0309 committed:
Merge pull request #1 from jiahui0309/branch-A-CodeQuality

Branch a code quality

jiahui0309 committed:
Fix checkstyle warnings for workflow

jiahui0309 committed:
Fix checkstyle warnings again

jiahui0309 committed:
BCD-Extension: Detect scheduling anomalies

jiahui0309 committed:
Fix bugs for date format

jiahui0309 committed:
Add user guide for the application

jiahui0309 committed:
Update README

jiahui0309 committed:
Minor fixes for release of application

jiahui0309 committed:
Fix bugs in build.gradle
